prow/pjutil/pjutil.go

```diff
@@ -42,14 +42,15 @@ func NewProwJob(spec prowapi.ProwJobSpec, extraLabels, extraAnnotations map[stri
 	labels, annotations := decorate.LabelsAndAnnotationsForSpec(spec, extraLabels, extraAnnotations)
 	specCopy := spec.DeepCopy()
 	setReportDefault(specCopy)
+	uuidV1, _ := uuid.NewUUID()
```

Comment:
Documentation reads:

> NewUUID returns a Version 1 UUID based on the current NodeID and clock sequence, and the current time. If the NodeID has not been set by SetNodeID or SetNodeInterface then it will be set automatically. If the NodeID cannot be set NewUUID returns nil. If clock sequence has not been set by SetClockSequence then it will be set automatically. If GetTime fails to return the current NewUUID returns nil and an error.
> In most cases, New should be used. 

However, `New()` will panic. Let's continue using `NewUUID()` but check for `nil` and an error.

Reply:
Thanks for review! This `pjutil.go` function `NewProwJob` is returning `prowapi.ProwJob`, and won't accept returning error message, err, or nil. I switched it to `New()`.

Same with the `horologium_test.go`, I think we can panic in tests if UUID generation fails.

`aksengine_helpers.go` is now returning error string.  

Reading off `satori/go.uuid` it really didn't handle errors anywhere, so I think if it had failed it would've panicked, so very little change in behavior I think?